### PR TITLE
Qute - fix validation bug

### LINKED ISSUE #15679: Qute fails to resolve a method invoked from an injected bean
## Describe the bug
Qute does not resolve a method properly to be invoked from an injected bean

## Expected behavior
The method in the injected bean is invoked

## Actual behavior
The following error is thrown during the build
```
io.quarkus.qute.TemplateException: Incorrect expression found: {inject:prettyTime.format(stats.updated)}
	- property/method [format(stats.updated)] not found on class [org.ocpsoft.prettytime.PrettyTime] nor handled by an extension method
	- at StatusResource/issues.html:17
		at io.quarkus.qute.deployment.QuteProcessor.processTemplateErrors(QuteProcessor.java:173)
```

## To Reproduce

1. Clone https://github.com/gastaldi/quarkus-status
2. Change to branch `pretty_time`
3. Run `mvn clean compile quarkus:dev`

## Configuration
```properties
# Add your application.properties here, if applicable.

```

## Screenshots
(If applicable, add screenshots to help explain your problem.)

## Environment (please complete the following information):
 - Output of `uname -a` or `ver`: Fedora 33
 - Output of `java -version`: JDK 11
 - GraalVM version (if different from Java): Not needed to reproduce
 - Quarkus version or git rev: 1.12.2.Final
 - Build tool (ie. output of `mvnw --version` or `gradlew --version`): Maven 3.6.3

### PR DESCRIPTION
**Pull request by mkouba**
- when a class declares multiple methods with the same name
- resolves #15679
- also process last part hints for virtual methods

### CODE REVIEW

File: extensions/qute/deployment/src/main/java/io/quarkus/qute/deployment/QuteProcessor.java [hunk 1]
```diff
@@ -1677,7 +1677,9 @@ private static AnnotationTarget findMethod(VirtualMethodPart virtualMethod, Clas
                         }
                         idx++;
                     }
-                    return matches ? method : null;
+                    if (matches) {
```
Comment (mkouba) [inline extensions/qute/deployment/src/main/java/io/quarkus/qute/deployment/QuteProcessor.java:1680]: @gastaldi FYI this is the fix for the problem described in the issue ;-).

**Review by gastaldi — APPROVED** [review]
LGTM

**Comment by gsmet** [comment]
I force pushed a rebase as this one was based on an old main which didn't have the latest CI changes.

**Comment by gsmet** [comment]
@famod I'm surprised by this run. It only touches Qute and triggered builds for everything. I rebased the PR so I'm surprised GIB is not working correctly.

Could you have a quick look? Maybe I did something wrong when renaming the branch.

**Comment by famod** [comment]
@gsmet I'm having a déjà-vu here because some few days ago there was another PR that "only touched Qute" and that triggered so many projects which surprised me as it surprised you!
Thing is: `Qute - Core` (`independent-projects/qute/core`) is used in `Quarkus - Dev tools - Codestarts` and from there it spreads over half the project.
A good way to check if GIB is doing something wrong is to use `-pl` and `-amd` yourself and voi­là, you get the same 493 modules that GIB built:
```
mvn validate -Dquickly -pl independent-projects/qute/core,extensions/qute/deployment -amd
[INFO] Scanning for projects...
[INFO] ------------------------------------------------------------------------
[...]
[INFO] ------------------------------------------------------------------------
[INFO] Reactor Build Order:
[INFO]
[INFO] Qute - Core                                                        [jar]
[INFO] Qute - Generator                                                   [jar]
[INFO] Quarkus - Dev tools - Codestarts                                   [jar]
[INFO] Quarkus - Dev tools - Common                                       [jar]
[INFO] Quarkus - Dev tools - Platform Descriptor - JSON Maven Plugin [maven-plugin]
[INFO] Quarkus - Platform Properties                                      [pom]
[INFO] Quarkus - Core - Deployment                                        [jar]
[INFO] Quarkus - JSON-P - Deployment                                      [jar]
[INFO] Quarkus - Test Framework - Common                                  [jar]
[INFO] Quarkus - Test Framework - JUnit 5 Internal Test Framework         [jar]
[...]
[INFO] Quarkus - Integration Tests - Google Cloud Function                [jar]
[INFO] Quarkus - Documentation                                            [jar]
[INFO]
[INFO] ---------------------< io.quarkus.qute:qute-core >----------------------
[INFO] Building Qute - Core 999-SNAPSHOT                                [1/493]
[INFO] --------------------------------[ jar ]---------------------------------
```
Btw, 493 modules are still less than those 800+ modules this repo has. 😉

Once it reaches the Test Framework it's kind of "game over" for a nice efficient incremental build.
Question is: Is there something to improve?

PS: It's actually funny what you can learn about the critical paths in a project after introducing incremental build. It happend to me more than once.

**Comment by famod** [comment]
I think that main problem is that `devtools-common` dependens on both `quarkus-devtools-codestarts` _and_ `qute-generator`.

`devtools-common` is in turn used by e.g. `platform-descriptor-json*` and from there you end up running all (or most?) ITs.

**Comment by famod** [comment]
In the meantime I'll merge this because CR1 is coming closer.

**Comment by aloubyansky** [comment]
> I think that main problem is that `devtools-common` depends on both `quarkus-devtools-codestarts` _and_ `qute-generator`.
> 
> `devtools-common` is in turn used by e.g. `platform-descriptor-json*` and from there you end up running all (or most?) ITs.
> 
> /cc @aloubyansky

We can and will definitely refine that. But the common tools will still be using Qute. We could also consider extracting Qute (core) as an independent project. Not sure though Martin will want to consider that. The tools are supposed to work with various Quarkus versions and so, probably, Qute.

**Comment by famod** [comment]
> But the common tools will still be using Qute. 

Well, if that cannot be changed, I don't see any real value in pulling out Qute core (in terms of incremental builds).

**Pull request merged**